tez-common/src/main/java/org/apache/tez/common/AsyncDispatcher.java

```diff
@@ -364,10 +369,20 @@ public void handle(Event event) {
       }
       try {
         eventQueue.put(event);
+        if (LOG.isTraceEnabled()) {
+          LOG.trace("AsyncDispatcher put event: {}", event);
+        }
       } catch (InterruptedException e) {
         if (!stopped) {
-          LOG.warn("AsyncDispatcher thread interrupted", e);
+          LOG.warn("AsyncDispatcher thread interrupted (while putting event): {}", event, e);
         }
+        // this is needed because in serviceStop we're looping to drain the rest of the events and checking !drained
+        // in that case, an interruption here would lead to an empty queue with drained=false, leading to
+        // a useless loop with a timeout (this can even cause unit test timeouts)
+        drained = eventQueue.isEmpty();
+        LOG.info(
+            "Setting drained to {} due to interruption while putting a new event, stopped: {}, blockNewEvents: {}",
+            drained, stopped, blockNewEvents);
```

Comment:
why in serviceStop we don't directly check ```eventQueue.isEmpty```?

Reply:
eh, right, let me try that :)

Reply:
I tried to look out for reasons for separate boolean, but didn't find anything reasonable, maybe the coding style then...